packages/x-data-grid-generator/package.json

```diff
@@ -38,7 +38,8 @@
     "@mui/x-data-grid-premium": "workspace:*",
     "chance": "^1.1.11",
     "clsx": "^2.1.1",
-    "lru-cache": "^7.18.3"
+    "lru-cache": "^7.18.3",
+    "msw": "^2.3.0"
```

Comment:
I am not so sure about using this, currently using it in 2 demos (the first demo on both overview and tree data sections). Here are a few pros and cons.

Pros:
- Monitor the data using Network tab
- Very close to the actual implementation

Cons:
- Not recommended to be used in Production
- Not meant for production so not very stable, need to avoid creating multiple instances of its server otherwise it behaves awkwardly

Suggestions are welcome.

Reply:
> Not recommended to be used in Production

Can you point me to the source? The definition of "production" is tricky for this kind of product 😅

> not very stable

Is this about the API stability, or something else?

Reply:
> Can you point me to the source?

https://deploy-preview-12317--material-ui-x.netlify.app/mockServiceWorker.js

> The definition of "production" is tricky for this kind of product 😅

Agreed, I couldn't find another evidence apart from the comment in mockServiceWorker.js and had its API stable, I would believe that the benefits outweigh the drawbacks.

> Is this about the API stability

Yes, you can not [have](https://github.com/mswjs/msw/discussions/821#discussioncomment-1023762) two instances of msw running concurrently, due to which we are left with the option to make it global for the docs package.

So I did an experiment in a new project by wrapping one instance in a `React.Context` provider and reusing it at multiple places within the application, but once we added more endpoints to it after the initial one, it behaved awkwardly, i.e. not providing the correct results.

I didn't dig deeper, we might be able to make it work by applying some hack, but I wanted to look into that as a follow-up, not in this PR, since it's not that simple. Wdyt?

Reply:
Another option could be to keep it running in one demo (tree data page first demo for now) as for one instance it's stable, and later improve the logic.

Reply:
> So I did an experiment in a new project by wrapping one instance in a React.Context provider and reusing it at multiple places within the application, but once we added more endpoints to it after the initial one, it behaved awkwardly, i.e. not providing the correct results.

Alright, this might be related to the Service Worker not being updated properly.
I had a bad experience with stale service workers in the past (https://github.com/mui/material-ui/issues/36808 to name one 😅).
If using msw produces confusing results, I would go with serving data from memory (so no requests in the Network tab), but with good logs in the console.

Reply:
Removed the msw for now and added console output in the demos.